cmake/modules/platform/toolcfg/gnu.cmake

```diff
@@ -87,20 +87,22 @@ endif()
 
 function(_omr_toolchain_separate_debug_symbols tgt)
 	set(exe_file "$<TARGET_FILE:${tgt}>")
-	set(dbg_file "$<TARGET_FILE:${tgt}>.dbg")
 	if(OMR_OS_OSX)
+		set(dbg_file "${exe_file}.dSYM")
 		add_custom_command(
 			TARGET "${tgt}"
 			POST_BUILD
-			COMMAND dsymutil -f ${exe_file} -o ${dbg_file}
+			COMMAND dsymutil -o "${dbg_file}" "${exe_file}"
 		)
 	else()
+		omr_get_target_path(target_path ${tgt})
+		omr_replace_suffix(dbg_file "${target_path}" ".debuginfo")
```

Comment:
do you think it would be cleaner (or possible) to use generator expressions?
maybe something like:

```cmake
(set dbg_file "$<TARGET_FILE_DIR:${tgt}>/$<TARGET_FILE_PREFIX:${tgt}>$<TARGET_FILE_BASE_NAME:${tgt}>.debuginfo")
```

Reply:
It doesn't appear that `TARGET_FILE_PREFIX` or `TARGET_FILE_BASE_NAME` are supported in cmake version 3.2 (the minimum required by OMR). Without assuming that the makefile generator is being used, I don't see how to express replacing the suffix.

Reply:
`TARGET_FILE_BASE_NAME` would not yield the desired result for OpenJ9 where the shared library suffix is often `29.so` (on Linux).

Reply:
realistically we probably want to bump omrs minimum cmake version to 3.4. Although that still doesn't help us with the TARGET_FILE_BASE_NAME which I don't believe was introduced until relatively recently (certainly after 3.10)

Reply:
It's not mentioned in the help for 3.14.7, but is documented in 3.15.7.